docs/user/using-an-OIDC-provider.md

```diff
@@ -0,0 +1,162 @@
+  template:
+    metadata:
+      labels:
+        name: kubeapps-auth-proxy
+    spec:
+      containers:
+      - args:
+        - --client-id=$AUTH_PROXY_CLIENT_ID
+        - --client-secret=$AUTH_PROXY_CLIENT_SECRET
+        - --discovery-url=$AUTH_PROXY_DISCOVERY_URL
+        - --skip-openid-provider-tls-verify
+        - --secure-cookie=false
+        - --upstream-url=http://kubeapps/
+        - --resources=uri=/api/kube/*|white-listed=true
+        - --listen=0.0.0.0:3000
+        image: keycloak/keycloak-gatekeeper
+        imagePullPolicy: IfNotPresent
+        name: kubeapps-auth-proxy
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: kubeapps-auth-proxy
+  name: kubeapps-auth-proxy
+spec:
+  ports:
+  - name: http
+    port: 3000
+    protocol: TCP
+    targetPort: 3000
+  selector:
+    name: kubeapps-auth-proxy
+  sessionAffinity: None
+  type: ClusterIP
+EOF
+```
+
+The above is a sample deployment, depending on the configuration of the Identity Provider those flags may vary. For this example we use:
+
+- `--client-id`, `--client-secret` and `--discovery-url`: Client ID, Secret and IdP URL as stated in the section above.
+- `--skip-openid-provider-tls-verify`, `--secure-cookie=false`: If the `discovery-url` is served through HTTPS with a self-signed certificate (i.e. Keycloak or Dex), those flags are necessary to avoid errors while validating the TLS certificate.
+- `--upstream-url`: Internal URL for the `kubeapps` service.
+- `--resources=uri=/api/kube/*|white-listed=true`: In order to use WebSockets we need to bypass the authentication for those request. In that case, Kubeapps will inject the `Bearer` token manually.
+- `listen=0.0.0.0:3000`: Listen in all the interfaces.
+
+## Exposing the proxy
+
+Once the proxy is in place and it's able to connect to the IdP we will need to expose it to access it as the main endpoint for Kubeapps (instead of the `kubeapps` service). We can do that with an Ingress object. Note that for doing so an [Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers) is needed. There are also other methods to expose the `kubeapps-auth-proxy` service, for example using `LoadBalancer` as type in a cloud environment. In case an Ingress is used, remember to modify the host `kubeapps.local` for the value that you want to use as hostname for your application:
+
+```
+kubectl create -n $KUBEAPPS_NAMESPACE -f - -o yaml << EOF
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  annotations:
+    nginx.ingress.kubernetes.io/connection-proxy-header: keep-alive
+    nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
+  name: kubeapps
+spec:
+  rules:
+  - host: kubeapps.local
+    http:
+      paths:
+      - backend:
+          serviceName: kubeapps-auth-proxy
+          servicePort: 3000
+        path: /
+EOF
+```
+
+Once you do that you can access the application, you will be prompted to introduce your credentials:
```

Comment:
I would make this more clear by saying something like, once you access the proxy you will be redirected to login with your Identity Provider. Once successfully logged in, you will be redirected to Kubeapps.